### PR TITLE
added test demonstrating the edge-case

### PR DESCRIPTION
**Pull request by vswarte**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ❌
| Fixed issues  | #18063

### CODE REVIEW

File: tests/framework/web/session/SessionTest.php [hunk 1]
```diff
@@ -88,4 +88,20 @@ public function testSetName()
...
+    {
+        $session = new Session();
+
+        $before = $session->getId();
+
+        $session->regenerateID();
+
+        $after = $session->getId();
+
+        $this->assertNotEquals($before, $after);
```
Comment (samdark) [inline tests/framework/web/session/SessionTest.php:105]: That's not what is documented in `regenerateID()` phpdoc.

Comment (marcovtwout-tremani) [reply tests/framework/web/session/SessionTest.php:105]: @samdark Note that documentation between https://github.com/yiisoft/yii2/blob/master/framework/web/Session.php#L287 and child class https://github.com/yiisoft/yii2/blob/master/framework/web/DbSession.php#L99 is not the same. (should it use inheritDoc?)

Comment (samdark) [reply tests/framework/web/session/SessionTest.php:105]: Yes, it should.
